drf_jsonapi/mixins.py

```diff
@@ -45,7 +44,8 @@ def list(self, request, *args, **kwargs):
             many=True,
             only_fields=request.fields,
             include=request.include,
-            page_size=self.request.GET.get('page[size]', settings.DEFAULT_PAGE_SIZE)
+            page_size=self.request.GET.get('page[size]', settings.DEFAULT_PAGE_SIZE),
```

Comment:
Since we're passing the request now, does it make sense to just extract the page size on the other side (in the serializer_class)? Also, why does this line reference `self.request` when we're passing in `request` as an argument to this method. Why do we have `request` and `self.request` in the same method?

Reply:
I think it makes sense to keep the fields, includes, and page_size discreet arguments so they *COULD* be independent of the request. We are passing the request now to allow for user-based filtering and validation of relationships (this follows the preferred pattern of DRF).

To your question about `self.request` vs `request`: I agree it should be `request`. I'll make that change. Django passes in the request AND sets it on the view instance so either would work but it should be consistent and using the value passed in is easier to test. Thanks for catching that!